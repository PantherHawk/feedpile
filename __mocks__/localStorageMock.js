const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"test","identities":[{"user_id":123}]}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
