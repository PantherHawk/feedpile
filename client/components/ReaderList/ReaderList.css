.reader-list-container {
  max-width: 600px;
  padding-right: 30px;
  padding-left: 280px;
  margin: auto;
}

@media (max-width: 600px) {
  .reader-list-container {
    padding-left: 30px;
  }
}
