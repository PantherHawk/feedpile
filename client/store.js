import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const composeSetup = process.env.NODE_ENV !== 'production' && typeof window === 'object' &&
window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;

const store = createStore(rootReducer, composeSetup (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
