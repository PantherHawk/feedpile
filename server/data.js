module.exports = {
 "users":[
  {
    "username": "MightyMouse",
    "password": "938dsweoirwu",
    "articles": [
    { "article_id": 0, "readFlag": true },
    { "article_id": 1, "readFlag": false }
    ]
  },
  {
    "username": "codybot@netscape.biz",
    "password": "938dsweoirwu",
    "articles": [
    { "article_id": 0, "readFlag": true },
    { "article_id": 1, "readFlag": false },
    { "article_id": 2, "readFlag": false }
  ]
  }
],
"feeds": [
  {
    "url": "techcrunch.com/feed/atom/",
    "name": "TechCrunch"
  },
  {
    "url": "feed.slate.com/slate",
    "name": "Slate"
  }
],
"articles": [
  {
    "article_url": "http://www.slate.com/articles/news_and_politics/jurisprudence/2017/04/how_the_supreme_court_will_change_after_the_nuclear_option.html",
    "date_written": "2017-04-08T22:16:01Z",
    "feedname": "Slate"
  },
  {
    "article_url": "https://techcrunch.com/2017/04/08/spotify-premium-director-robert-lamvik-leaves-company-for-meditation-app-headspace/",
    "date_written": "2017-04-08T16:13:43Z",
    "feedname": "TechCrunch"
  },
  {
    "article_url": "https://techcrunch.com/2017/04/08/spotify-premium-director-robert-lamvik-leaves-company-for-meditation-app-headspace/",
    "date_written": "2017-04-07T21:16:01Z",
    "feedname": "TechCrunch"
  },
 {
    "article_url": "https://techcrunch.com/2017/04/08/spotify-premium-director-robert-lamvik-leaves-company-for-meditation-app-headspace/",
    "date_written": "2017-04-07T21:16:01Z",
    "feedname": "TechCrunch"
 }
]

}