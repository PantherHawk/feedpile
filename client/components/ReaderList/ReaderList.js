import React from 'react';
import ReaderListHeader from '../ReaderListHeader/ReaderListHeader'
import ReaderListItem from '../ReaderListItem/ReaderListItem';
import {connect} from 'react-redux';
import './ReaderList.css';
import { fetchArticlesForFeedsFromDb } from '../../actions'

class ReaderList extends React.Component {
  constructor (props){
    super(props);
  }
  componentWillMount() {
    this.props.fetchArticlesForFeedsFromDb();
  }
  render() {
    let articlesFound = [];
    if(!Array.isArray(this.props.articles)) {
      if(this.props.currentFeed !== "") {
        articlesFound = this.props.articles[this.props.currentFeed];
      } else {
        for(var key in this.props.articles) {
          articlesFound = articlesFound.concat(this.props.articles[key]);
        }
      }
    }
    return (
        <div className="reader-list-container">
          <ReaderListHeader />
          {articlesFound.map((article, index) => (
              <ReaderListItem history={this.props.history}
                              article={article}
                              key={index} articleIndex={index}
              />
          ))}
        </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    articles: state.articles,
    currentFeed: state.currentFeed
  }
};

export const Unwrapped = ReaderList;
export default connect(mapStateToProps, {fetchArticlesForFeedsFromDb})(ReaderList);
