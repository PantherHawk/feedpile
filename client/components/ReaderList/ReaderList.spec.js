import '../../../__mocks__/localStorageMock'
import React from 'react';
import {Provider} from 'react-redux';
import store from '../../store'
import ReaderList, {Unwrapped as UnwrappedReaderList} from './ReaderList';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import data from '../../../data.json'

const tempData = {
  "58f7a4b09f861f7da0e08542": [
    {
      "_id": "58f82c8410b7be21d128551c",
      "__v": 0,
      "title": "Cycling to work can cut cancer and heart disease, says study",
      "url": "http://www.bbc.co.uk/news/health-39641122",
      "imageSrc": "http://c.files.bbci.co.uk/16C69/production/_95698239_gettyimages-168396725.jpg",
      "date": "2017-04-20T03:10:41.000Z",
      "datePublished": "2017-04-20T03:10:41.000Z",
      "feedId": "58f7a4b09f861f7da0e08542",
      "summary": "Study of 250,000 UK commuters shows walking is good too but suggests two wheels\nare best.",
      "description": "Study of 250,000 UK commuters shows walking is good too but suggests two wheels are best."
    },
    {
      "_id": "58f82d39e3ad80221d881545",
      "__v": 0,
      "title": "Experts excited by brain 'wonder-drug'",
      "url": "http://www.bbc.co.uk/news/health-39641123",
      "imageSrc": "http://c.files.bbci.co.uk/85F1/production/_95698243_gettyimages-535164801.jpg",
      "date": "2017-04-20T03:05:26.000Z",
      "datePublished": "2017-04-20T03:05:26.000Z",
      "feedId": "58f7a4b09f861f7da0e08542",
      "summary": "A drug for depression could stop all neurodegenerative diseases, including\ndementia, scientists hope.",
      "description": "A drug for depression could stop all neurodegenerative diseases, including dementia, scientists hope."
    }
  ]
}

test('ReaderList take a snapshot', () => {
  const component = shallow(<UnwrappedReaderList fetchArticlesForFeedsFromDb={()=>{}} articles={data}/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
test('ReaderList should render a ReaderListItem for each article', () => {

  const component = render(<Provider store={store}><UnwrappedReaderList fetchArticlesForFeedsFromDb={()=>{}} articles={tempData} currentFeed={""}/></Provider>);
  expect(component.find('.list-item-container').length).toEqual(tempData["58f7a4b09f861f7da0e08542"].length);
});
