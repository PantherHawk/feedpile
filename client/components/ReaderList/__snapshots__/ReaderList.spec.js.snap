// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReaderList take a snapshot 1`] = `
<div
  className="reader-list-container"
>
  <Connect(ReaderListHeader) />
</div>
`;
